src/backend/InvenTree/InvenTree/templatetags/inventree_extras.py

```diff
@@ -663,3 +663,43 @@ def admin_url(user, table, pk):
             pass
 
     return url
+
+
+@register.simple_tag()
+def approval_allowed(user, order):
```

Comment:
I would not put this code into "inventree_extras" as a template tag. This should be a method on the `PurchaseOrder` model, such as:

```python
def approval_allowed(self, user):
    ...
```

And then *call* that from the template tag. That will make the code far more re-usable.

Reply:
Done